File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.cpp
class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
        vector<int> ans;
        map<int, int> result;
        
        for(int i = 0; i < matrix.size(); i++){
            int min = (*min_element(matrix[i].begin(), matrix[i].end()));
            for(int j = 0; j < matrix[i].size(); j++){
                if(matrix[i][j] == min){
                    result[matrix[i][j]] = j;
                }
            }
        }
        
        bool flag = true;
        for(auto p : result){
            flag = true;
            for(int i = 0; i < matrix.size(); i++){
                if(matrix[i][p.second] > p.first) { flag = false; break; }
            }
            if(flag) ans.push_back(p.first);
        }
        
        return ans;
    }
};

File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        int[] rmin = new int[m];
        for (int i = 0; i < m; i++) {
            int min = matrix[i][0];
            for (int j = 1; j < n; j++) {
                if (matrix[i][j] < min) {
                    min = matrix[i][j];
                }
            }
            rmin[i] = min;
        }
        int[] cmax = new int[n];
        for (int j = 0; j < n; j++) {
            int max = matrix[0][j];
            for (int i = 1; i < m; i++) {
                if (matrix[i][j] > max) {
                    max = matrix[i][j];
                }
            }
            cmax[j] = max;
        }
        List<Integer> arr = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (matrix[i][j] == rmin[i] && matrix[i][j] == cmax[j]) {
                    arr.add(matrix[i][j]);
                }
            }
        }
        return arr;
    }
}